app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -128,25 +128,47 @@ export default class RnsResolver extends RnsJsDelegate {
     });
   }
 
-  // this.rifConfig.rns.contracts.multiChainResolver This is the path to a multicahin address in the config
-  // TODO: Rodrigo
-  getNetoworksForResolvers (address) {
+  /**
+   * Returns an array of network addresses for a given domain
+   * @param domainNameResolver DomainName with the .rsk extension
+   * @returns {Promise<unknown>}
+   */
+  getNetworksForResolvers (domainNameResolver) {
     return new Promise((resolve, reject) => {
+      // TODO: Rodrigo
       // This event is different, it deppends if we're on the chain of RSK, or not, to develop purposes i'll use only ChainAddrChanged, it need to differentiate it from one to another
-      console.debug('INICIO');
-
-      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash('testing.rsk')}, {fromBlock: 0, toBlock: 'latest'});
+      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainNameResolver)}, {fromBlock: 0, toBlock: 'latest'});
       myEvent.get(function(error, result){
-        console.debug('RESULT ===================', result);
-        console.debug('ERROR ===================', error);
+        if (error) {
+          console.debug('Error when trying to get netoworks for resolver', error);
+          reject(error);
+        }
+        const arrChains = [];
+        result.forEach(network => {
+          const networkToPush = new ResolverNetwork(network.args.chain, network.args.addr)
+          const index = arrChains.findIndex((e) => e.chain === networkToPush.chain);
+          if (index === -1) {
+            arrChains.push(networkToPush);
+          } else {
+            arrChains[index] = networkToPush;
+          }
+        });
+        console.debug('getNetworksForResolvers success', arrChains);
+        resolve(arrChains);
       });
-      console.debug('FIN');
     });
   }
 
-  setNetoworksForResolver () {
+  /**
+   * Calls the contract and sets a new network calling the setChainAddr function in the contract of multichainresolver
+   * @param domainNameResolver domainNameResolver DomainName with the .rsk extension
+   * @param chain
+   * @param networkAddress
+   * @returns {Promise<unknown>}
+   */
+  setNetoworksForResolver (domainNameResolver, chain, networkAddress) {
```

Comment:
I dont find the function name nor the documentation about this method related with its behaviour. 

This invokes `setChainAddr`, and that set an address into the chain map of the domain node.
```

 function setChainAddr (bytes32 node, bytes4 chain, string memory addrValue) public onlyOwner(node) {
        chainAddresses[node][chain].addr = addrValue;
        if (chain == RSK_CHAIN_ID) {
            address _addr = addressHelper.stringToAddress(addrValue);
            emit AddrChanged(node, _addr);
        } else {
            emit ChainAddrChanged(node, chain, addrValue);
        }
    }
```

Reply:
Changed in previous commit!